python/kaon_physics.py

```diff
@@ -0,0 +1,109 @@
+
+class KaonPhysics():
+    """Parameters that determine the physics of kaons in the simulation
+
+    Parameters
+    ----------
+    kplus_branching_ratios : list[float]
+    kminus_branching_ratios : list[float]
+    k0l_branching_ratios : list[float]
+    k0s_branching_ratios : list[float]
+        List of the desired branching ratios for kaons. The entries
+        correspond go the following processes
+
+        For charged kaons
+        - 0: K -> mu + v
+        - 1: K -> pi + pi0
+        - 2: K -> 3 pi
+        - 3: K -> pi + 2 pi0
+        - 4: K -> pi0 + e + v
+        - 5: K -> pi0 + mu + v
+
+        For Klong
+        - 0: K -> 3 pi0
+        - 1: K -> pi0 + pi+ + pi-
+        - 2: K -> pi- + e+ + v
+        - 3: K -> pi+ + e- + v
+        - 4: K -> pi- + mu+ + v
+        - 5: K -> pi+ + mu- + v
+
+        For Kshort:
+        - 0: K -> pi+ + pi-
+        - 1: K -> 2 pi0
+
+        See the sources for Geant4's definitions and branching ratios
+        (G4KaonMinus.cc, G4KaonPlus.cc, KaonZeroLong.cc, KaonZeroShort.cc). The
+        default branching ratios correspond to the Geant4 defaults.
+
+    kplus_lifetime_factor : float
+    kminus_lifetime_factor : float
+    k0l_lifetime_factor : float
+    k0s_lifetime_factor : float
+
+        Multiplicative factor to scale the lifetime of kaons by. Default is to
+        scale by 1 (no scaling)
+
+    """
+    def __init__(self):
+        self.kplus_branching_ratios = [
+            0.6355,  # K^+ -> mu^+ + nu_mu
+            0.2066,  # K^+ -> pi^+ + pi^0
+            0.0559,  # K^+ -> pi^+ + pi^- + pi^+
+            0.01761, # K^+ -> pi^+ + pi^0 + pi^0
+            0.0507,  # K^+ -> pi^0 + e^+ + nu_e
+            0.0335,  # K^+ -> pi^0 + mu^+ + nu_mu
+        ]
+        self.kminus_branching_ratios = [
+            0.6355,  # K^- -> mu^- + anti_nu_mu
+            0.2066,  # K^- -> pi^- + pi^0
+            0.0559,  # K^- -> pi^- + pi^+ + pi^-
+            0.01761, # K-+ -> pi^- + pi^0 + pi^0
+            0.0507,  # K-+ -> pi^0 + e^- + anti_nu_e
+            0.0335,  # K-+ -> pi^0 + mu^- + anti_nu_mu
+        ]
+        self.k0l_branching_ratios = [
+            0.1952, # K^0_L -> pi^0 + pi^0 + pi^0
+            0.1254, # K^0_L -> pi^0 + pi^+ + pi^-
+            0.2027, # K^0_L -> pi^- + e^+ + nu_e
+            0.2027, # K^0_L -> pi^+ + e^- + anti_nu_e
+            0.1352, # K^0_L -> pi^- + mu^+ + nu_mu
+            0.1352, # K^0_L -> pi^+ + mu^- + anti_nu_mu
+        ]
+        self.k0s_branching_ratios = [
+            0.6920, # K^0_S -> pi^+ + pi^-
+            0.3069, # K^0_S -> pi^0 + pi^0
+        ]
+        self.kplus_lifetime_factor = 1.
+        self.kminus_lifetime_factor = 1.
+        self.k0l_lifetime_factor = 1.
+        self.k0s_lifetime_factor = 1.
+
+
+    def upKaons():
+        """Returns a configuration of the kaon physics corresponding
+        to the changes that were made in
+        https://github.com/ldmx-software/geant4/tree/LDMX.upKaons_mod
```

Comment:
This isn't actually asking for more in-code comments, but I am curious - what is corresponding in this context? Did you copy the values over or did you do more checking against the upKaon compiled branch of geant4?

Reply:
I went through the commits and checked all the kaon particle-related changes (i.e. everything that isn't to do with the bertini cascade). The only changes are here
https://github.com/LDMX-Software/geant4/commit/25228b8b1fbad913b4933b7c0d3951ebe36d404c 

The values in the config _should_ correspond to the changes here (factor 1/50 in lifetime, changing all non-leptonic branching ratios to 0 and scaling up the others)

Reply:
Sounds good, thanks for the extra detail